### PR TITLE
ENT-12435: Fixed openssl build on hpux and tidied up unused patches

### PR DESCRIPTION
**Pull request by craigcomstock**
- Removed patches no longer needed for openssl on solaris
- Removed unused patch for openssl on aix
- Removed patch for hpux openssl that is no longer needed

[Build Status](https://ci.cfengine.com/job/pr-pipeline/11400/)

### CODE REVIEW

**Comment by craigcomstock** [comment]
@larsewi @vpodzime I tested a package for a while. Bootstrapping, reporting, they all seemed to work fine.

http://buildcache.cloud.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11400/PACKAGES_ia64_hpux_11.23/

Look at our hub if you are curious. I bootstrapped it there and so there is still a record there that I intend on using periodically.

**Review by larsewi — APPROVED** [review]
:rocket: :beers:

**Comment by craigcomstock** [comment]
pick needed to 3.24.x: https://github.com/cfengine/buildscripts/pull/1531

File: deps-packaging/openssl/hpux/build [hunk 1]
```diff
@@ -10,7 +10,6 @@ OSD=${BUILD_ROOT}/cfbuild-openssl-devel${PREFIX}
 # To pick up libgcc_s.so.1, which is apparently not in dyld path.
 export LD_LIBRARY_PATH=$PREFIX/lib
 
-$PATCH -p1 < 0010-Revert-Add-exemplar-use-case-for-rcu-locks.patch
```
Comment (vpodzime) [inline deps-packaging/openssl/hpux/build:13]: Does it mean RCU locks work on HP-UX now? Or are they disabling them upstream now?

**Pull request merged**